spyder/app/start.py

```diff
@@ -151,6 +151,8 @@ def main():
     if options.debug_info:
         levels = {'minimal': '2', 'verbose': '3'}
         os.environ['SPYDER_DEBUG'] = levels[options.debug_info]
+    elif CONF.get('main', 'startup_in_debug_mode'):
+        os.environ['SPYDER_DEBUG'] = '3'

```

Comment:
I don't like this one bit. The debugging state shouldn't be set in our Preferences but instead using environment variables in our `app/restart.py` script. Else Spyder could be left in debug mode forever if users forget to change the value of that option.

Reply:
Okay. I will rework this.